src/java/org/apache/cassandra/index/sai/disk/vector/CassandraDiskAnn.java

```diff
@@ -232,11 +237,12 @@ public CloseableIterator<RowIdWithScore> search(VectorFloat<?> queryVector,
             if (features.contains(FeatureId.FUSED_ADC))
             {
                 var asf = view.approximateScoreFunctionFor(queryVector, similarityFunction);
-                var rr = view.rerankerFor(queryVector, similarityFunction);
+                var rr = !isRerankless ? view.rerankerFor(queryVector, similarityFunction) : null;
```

Comment:
Nit: we can do this without the negation
```suggestion
                var rr = isRerankless ? null : view.rerankerFor(queryVector, similarityFunction);
```

Reply:
Yeah, I thought about this and I'll change it, but I really like having the null at the end of the line and having the variable name as `isRerankless`